File: order-service/src/main/java/com/catius/order/controller/dto/OrderResponse.java
package com.catius.order.controller.dto;

import com.catius.order.domain.Order;

import java.time.Instant;
import java.util.List;

public record OrderResponse(
        String orderId,
        long customerId,
        String status,
        List<OrderItemResponse> items,
        Instant createdAt
) {

    public record OrderItemResponse(long productId, int quantity) {
    }

    public static OrderResponse from(Order order) {
        List<OrderItemResponse> items = order.getItems().stream()
                .map(i -> new OrderItemResponse(i.getProductId(), i.getQuantity()))
                .toList();
        return new OrderResponse(
                order.getOrderId(),
                order.getCustomerId(),
                order.getStatus().name(),
                items,
                order.getCreatedAt()
        );
    }
}

File: order-service/src/main/kotlin/com/catius/order/controller/dto/OrderResponse.kt
package com.catius.order.controller.dto

import com.catius.order.domain.usecase.command.OrderResult
import java.time.Instant

data class OrderResponse(
    val id: Long,
    val customerId: Long,
    val items: List<OrderItemResponse>,
    val status: String,
    val createdAt: Instant,
    val updatedAt: Instant,
) {
    companion object {
        fun from(result: OrderResult): OrderResponse {
            return OrderResponse(
                id = result.id,
                customerId = result.customerId,
                items = result.items.map { OrderItemResponse.from(it) },
                status = result.status.name,
                createdAt = result.createdAt,
                updatedAt = result.updatedAt
            )
        }
    }
}

data class OrderItemResponse(
    val productId: Long,
    val quantity: Int,
) {
    companion object {
        fun from(item: OrderResult.Item): OrderItemResponse {
            return OrderItemResponse(
                productId = item.productId,
                quantity = item.quantity
            )
        }
    }
}
